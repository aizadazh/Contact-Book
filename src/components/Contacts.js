import React, { Component, Fragment } from "react";
import styled from "styled-components";

const Input = styled.input`
    width: 10em;
    height: 3em;
    border-radius: 3px;
    border: 1px solid grey;
    &:placeholder{
        color: grey;
    }
`;

export default class Contacts extends Component {
    render(){
        return (
            <Fragment>
                <Input type="text" name="text" placeholder="Name"></Input>
                <Input type="text" name="text" placeholder="Surname"></Input>
                <Input type="text" name="text" placeholder="Phone number"></Input>
                <button type="submit">Add contact</button>
            </Fragment>
        )
    }
}