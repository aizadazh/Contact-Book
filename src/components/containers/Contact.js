import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { deleteContact } from "../../actions/ContactAction";


export class Contact extends Component {
    state = {
        showContactInfo: false,
    };
    onDeleteClick  = id => {
        this.props.deleteContact(id);
    };
    render(){
        const { id, name, address, phone } = this.props.contact;
        console.log(this.props.contact.address.street)
        const { showContactInfo } = this.state;
        let nameOfStreet=''
        if(this.props.contact.address.substring){
            nameOfStreet=address
        }else{
            nameOfStreet=address.street
        }
        return(
            <div>
                <h4>
                    {name}{" "}
                    <i 
                        onClick = {() =>
                            this.setState({
                                showContactInfo: !this.state.showContactInfo,
                            })
                        }
                        className = "fas fa-sort-down"
                        style = {{cursor: "pointer", color: "blue"}}
                        />
                    <i 
                        className = "fas fa-times"
                        style = {{cursor: "pointer", float:"right", color: "red"}} 
                        onClick = {this.onDeleteClick.bind(this, id)}
                    />
                    <Link to ={`contact/edit/${id}`}>
                        <i
                            className = "fas fa-pencil-alt"
                            style={{
                                cursor: "pointer",
                                float: "right",
                                color: "green",                               
                                marginRight: "1rem",
                                fontSize: "0.7em",
                                paddingTop: "0.3em",
                            }}
                        />
                    </Link>                    
                </h4>
                    {showContactInfo ? (
                        <ul className="list-group">
                            <li className="list-group-item">Address: {nameOfStreet}</li>
                            <li className="list-group-item">Phone: {phone}</li>
                        </ul>
                    ) : null }
            </div>
        );
    }
}
Contact.propTypes = {
    contact: propTypes.object.isRequired,
    deleteContact: propTypes.func.isRequired,
};
export default connect(
    null, 
    { deleteContact },    
)(Contact);
