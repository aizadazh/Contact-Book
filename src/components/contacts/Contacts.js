import React, { Component, Fragment } from "react";
import Contact from "./Contact";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { getContacts } from "../../actions/ContactAction";

class Contacts extends Component {
  componentDidMount() {
    this.props.getContacts();
  }
  render() {    
    const { contacts } = this.props;
    return (
      <Fragment>
        <h1 className="display-5 mb-2">
          <span className="text-danger">Список контактов</span>
        </h1>
        {contacts.map(contact => (
          <Contact key={contact.id} contact={contact} />
        ))}
      </Fragment>
    );
  }
}
Contacts.propTypes = {
  contacts: propTypes.array.isRequired,
  getContacts: propTypes.func.isRequired,
};

const mapStateToProps = state => ({
  contacts: state.contact.contacts,
});

export default connect(
  mapStateToProps,
  { getContacts },
)(Contacts);

