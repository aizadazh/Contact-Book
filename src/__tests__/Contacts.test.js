import React from "react";
import { shallow } from "enzyme";
import { Contacts } from "../components/containers/Contacts";


describe("Testing Contacts component", () => {
    const props = {
        contacts: [],
        getContacts: jest.fn()
    }
describe("Testing the component is rendered correctly", () => {
    const wrapper = shallow(<Contacts {...props} />)
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();        
        expect(wrapper.find(".text-danger").text()).toBe("Список контактов")
    })
});
    
});