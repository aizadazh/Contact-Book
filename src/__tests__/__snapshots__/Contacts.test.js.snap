// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Contacts component Testing the component is rendered correctly renders correctly 1`] = `ShallowWrapper {}`;
